course/page/code.py

```diff
@@ -339,34 +338,7 @@ def filter_from_code_html_attributes(tag, name, value):
         else:
             return False
 
-    else:
-        # Approach recommended here: https://github.com/mozilla/bleach/issues/348
-
-        # FIXME: Cannot access 'namespaced name' of attribute, so
-        # we're unable to match XML's base and xlink.
-        if (None, name) in attr_val_is_uri:
-            # Reluctant copy-paste from
-            # https://github.com/mozilla/bleach/blob/8706b5373b633407e804b1b2975edf8760067ff7/bleach/sanitizer.py#L513-L525
-
-            from xml.sax.saxutils import unescape
-
-            val_unescaped = re.sub(
-                "[`\000-\040\177-\240\s]+",
-                '',
-                unescape(value)).lower()
-
-            # Remove replacement characters from unescaped characters
-            val_unescaped = val_unescaped.replace("\ufffd", "")
-
-            # Drop attributes with uri values that have
-            # protocols that aren't allowed
-            if (re.match(r'^[a-z0-9][-+.a-z0-9]*:', val_unescaped)
-                    and
-                    (val_unescaped.split(':')[0]
-                        not in bleach.ALLOWED_PROTOCOLS)):
-                return False
-
-        return name in bleach.ALLOWED_ATTRIBUTES
+    return False
```

Comment:
Doesn't this mean that we delete attributes that `bleach` would normally allow?

Reply:
No.

Reply:
I think it acts like removing items from blacklist, not a whitelist.

Reply:
In terms of attribute, I might be wrong, I'll check that.

Reply:
Whether keeping those snippet or not, `bleach`'s default allowed tag-attributes are not allowed. Because that had been overriden by the callback `filter_from_code_html_attributes`.

`bleach`'s default allowed attributes are at [`here`](https://github.com/mozilla/bleach/blob/master/bleach/sanitizer.py#L51).